cvmfs/sync_mediator.cc

```diff
@@ -993,14 +1050,14 @@ void SyncMediator::AddLocalHardlinkGroups(const HardlinkGroupMap &hardlinks) {
     }
 
     if (params_->print_changeset) {
-      std::string changeset_notice = "add hardlinks around ("
+      std::string changeset_notice = "Hardlinks around ("
                                    + i->second.master->GetUnionPath() + ")";
       for (SyncItemList::const_iterator j = i->second.hardlinks.begin(),
            jEnd = i->second.hardlinks.end(); j != jEnd; ++j)
       {
         changeset_notice += " " + j->second->filename();
       }
-      PrintChangesetNotice(kAddHardlinks, changeset_notice);
+      reporter_->OnAdd(changeset_notice, catalog::DirectoryEntry());
```

Comment:
In the light of our general strategy to break up hardlinks, we should change this a little.  Instead of telling the hardlink group, let's treat the hardlinks like individual files. That means one `OnAdd()` call for `i->second.master` and then a call for every entry from `i->second.hardlinks` in the `for` loop.  In the for loop, you need to reconstruct the full path like

    GetParentPath(i->second.master->GetUnionPath()) + "/" + j->second->filename()